### ISSUE TITLE
catshadow client_docker_test failure due to retransmission

### ISSUE THREAD

**Issue by mixmasala** [katzenpost/katzenpost#152]
```
17:14:47.377 DEBU catshadow: Saving statefile.
17:14:47.385 DEBU catshadow: readInbox command d80c9f2d221ce74897213c56afbc03d1 sent
    client_docker_test.go:764: Message received by b2
    client_docker_test.go:776: 
        	Error Trace:	/go/katzenpost/catshadow/client_docker_test.go:776
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestDockerRenameContact
--- FAIL: TestDockerRenameContact (100.41s)
17:14:47.394 DEBU 
```

The test is just counting received messages, and a retransmission due to lost surb-ack causes the number of total messages received to not match the expected number.

**Comment by leif** [comment]
different but related failure, from CI today:
```
2023-01-03T00:47:05.2737920Z 00:47:05.273 INFO provider2_client: OnACK with SURBID [6470d1e30984890b13bedf0ba9c695c3]
2023-01-03T00:47:05.2874683Z 00:47:05.286 DEBU provider2_client: doSend [28d70a0fc9b519e4ce517d333f1da8bc] without SURB
2023-01-03T00:47:05.2875499Z 00:47:05.286 DEBU minclient:de7600004425a919eb7b6f45c47d99c2c97c712d47061dcf87c8d0a8ff02b1ce@provider2: Hop[0] 'provider2' - 410 ms
2023-01-03T00:47:05.2876135Z 00:47:05.286 DEBU minclient:de7600004425a919eb7b6f45c47d99c2c97c712d47061dcf87c8d0a8ff02b1ce@provider2: Hop[1] 'mix1' - 91 ms
2023-01-03T00:47:05.2876721Z 00:47:05.286 DEBU minclient:de7600004425a919eb7b6f45c47d99c2c97c712d47061dcf87c8d0a8ff02b1ce@provider2: Hop[2] 'mix2' - 216 ms
2023-01-03T00:47:05.2877327Z 00:47:05.286 DEBU minclient:de7600004425a919eb7b6f45c47d99c2c97c712d47061dcf87c8d0a8ff02b1ce@provider2: Hop[3] 'mix3' - 156 ms
2023-01-03T00:47:05.2877935Z 00:47:05.286 DEBU minclient:de7600004425a919eb7b6f45c47d99c2c97c712d47061dcf87c8d0a8ff02b1ce@provider2: Hop[4] 'provider2' - 0 ms
2023-01-03T00:47:05.3232712Z 00:47:05.321 DEBU minclient/conn:de7600004425a919eb7b6f45c47d99c2c97c712d47061dcf87c8d0a8ff02b1ce@provider2: Dequeued packet for send.
2023-01-03T00:47:05.3240910Z 00:47:05.323 DEBU minclient/conn:5d044731a32c91741344578b41f76f4afcb709a3f0072bee62d725dc63bf810b@provider2: Sent RetrieveMessage: 40
2023-01-03T00:47:05.3245951Z 00:47:05.323 DEBU minclient/conn:5d044731a32c91741344578b41f76f4afcb709a3f0072bee62d725dc63bf810b@provider2: Received MessageACK: 40
2023-01-03T00:47:05.3246567Z 00:47:05.323 INFO provider2_client: OnACK with SURBID [db37e36c9e3d0335f3b6214eea0328b3]
2023-01-03T00:47:05.3247471Z 00:47:05.324 DEBU catshadow: MessageDeliveredEvent for b2 MessageID 0fd5e5645498555e803db4d21f6fd9b8
2023-01-03T00:47:05.3247845Z 00:47:05.324 DEBU catshadow: Sending MessageDeliveredEvent for b2
2023-01-03T00:47:05.3248151Z 00:47:05.324 DEBU catshadow: Saving statefile.
2023-01-03T00:47:05.3289245Z 00:47:05.328 DEBU catshadow: No messages to send for contact: b2
2023-01-03T00:47:05.3296234Z     client_docker_test.go:751: Message delivered to b2
2023-01-03T00:47:05.3302545Z     client_docker_test.go:765: Message received by b2
2023-01-03T00:47:05.3303280Z     client_docker_test.go:802: 
2023-01-03T00:47:05.3306482Z         	Error Trace:	/go/katzenpost/catshadow/client_docker_test.go:802
2023-01-03T00:47:05.3307017Z         	Error:      	Not equal: 
2023-01-03T00:47:05.3307240Z         	            	expected: 1
2023-01-03T00:47:05.3307452Z         	            	actual  : 2
2023-01-03T00:47:05.3307689Z         	Test:       	TestDockerRenameContact
2023-01-03T00:47:05.3308219Z --- FAIL: TestDockerRenameContact (63.17s)
2023-01-03T00:47:05.3308456Z FAIL
```

**Comment by leif** [comment]
this should be fixed by #223.
